### ISSUE TITLE
DS-Net for object detection 

### ISSUE THREAD

**Issue by NoLookDefense** [changlin31/DS-Net#14]
Hello. Thanks for your work. I noticed that you also conducted some experiments in object detection. I wonder whether or when you will release the code

**Comment by changlin31** [comment]
Hi, @NoLookDefense 
Thanks for your interest. Our experiments on object detection is conducted base on another Github repo (https://github.com/lzx1413/PytorchSSD). Please also refer to issue https://github.com/changlin31/DS-Net/issues/4

**Comment by NoLookDefense** [comment]
Thank you. 
I noticed that the code is recently updated two years ago. Is it just an SSD implementation, without your algorithms like slimmable NN and gating implementation?

**Comment by changlin31** [comment]
Yes, we currently don't have plans to release the detailed code for object detection. Sorry for any inconvenience.

**Comment by twmht** [comment]
@changlin31 

Do we need to use the pretrained weight trained from slimmer algorithm if we want to train object detection?

**Comment by changlin31** [comment]
> @changlin31
> 
> Do we need to use the pretrained weight trained from slimmer algorithm if we want to train object detection?

Hi @twmht 

We used the dynamic slimmable pretrained weight for our object detection results in the paper. But I expect performing slimmable training on object detection task with a normally pretrained network is also possible. (We are getting good results with the second training approach on action recognition tasks)

**Comment by twmht** [comment]
@changlin31 

What is your second training approach? Did you mention in the paper?

**Comment by changlin31** [comment]
@twmht 

By second approach, I mean: **first**, directly load a normal ImageNet pre-trained model (e.g. pytorch ResNet checkpoint) and **then**, perform supernet training and gate training on the downstream tasks.

I did not mention this in the paper. This is a work we are currently working on.

**Comment by twmht** [comment]
@changlin31 

did you also use in-place distilling with object detection? Which loss did you use for the classification branch?
